rotkehlchen/externalapis/etherscan.py

```diff
@@ -236,6 +240,47 @@ def _query(
 
         return result
 
+    @staticmethod
+    def _get_or_create_genesis_transaction(
+            database: DBHandler,
+            account: ChecksumEthAddress,
+    ) -> EthereumTransaction:
+        """Add 0x0 txn to DB if it is missing"""
+        db_ethx = DBEthTx(database)
+
+        # Fetch the genesis txn from DB
+        result = db_ethx.get_ethereum_transactions(
```

Comment:
As I mentioned before there are methods that handle writing transactions to the database. Take a look [here](https://github.com/rotki/rotki/blob/develop/rotkehlchen/chain/ethereum/transactions.py#L250). This is a place where parent transaction is added to the database if it is not there yet. Why don't you move this code to that place?